Testscripts/Linux/STOR-Lis-Disk.sh

```diff
@@ -0,0 +1,184 @@
+            LogMsg "Checksum mismatch on  block $y for ${targetDevice} "
+            SetTestStateFailed
+            exit 0
+        fi
+    done
+    LogMsg "Data integrity test on ${blocks} blocks on drive $1 : success "
+    umount /mnt/
+    rm -f $testFile
+}
+
+SetTestStateRunning
+# Count the number of SCSI= and IDE= entries in constants
+#
+diskCount=1
+# Set to 1 because of resource disk
+for entry in $(cat ./constants.sh); do
+    # Convert to lower case
+    lowStr="$(tr '[A-Z]' '[a-z' <<<"$entry")"
+
+    # does it start wtih ide or scsi
+    if [[ $lowStr == ide* ]]; then
+        diskCount=$((diskCount + 1))
+    fi
+
+    if [[ $lowStr == scsi* ]]; then
+        diskCount=$((diskCount + 1))
+    fi
+done
+
+LogMsg "constants disk count = $diskCount"
+
+#
+# Compute the number of sd* drives on the system.
+#
+sdCount=0
+for drive in /dev/sd*[^0-9]; do
+    sdCount=$((sdCount + 1))
+done
+
+#
+# Subtract the boot disk from the sdCount, then make
+# sure the two disk counts match
+#
+sdCount=$((sdCount - 1))
+LogMsg "/dev/sd* disk count = $sdCount"
+
+if [ $sdCount != $diskCount ]; then
+    LogMsg " disk count ($diskCount) does not match disk count from /dev/sd* ($sdCount)"
+    SetTestStateAborted
+    exit 0
+fi
+
+#
+# For each drive, run fdisk -l and extract the drive
+# size in bytes.  The setup script will add Fixed
+#.vhd of size 1GB, and Dynamic .vhd of 137GB
+#
+FixedDiskSize=1073741824
+Disk4KSize=4096
+DynamicDiskSize=136365211648
+
+for driveName in /dev/sd*[^0-9]; do
+    # Skip /dev/sda and /dev/sdb
+    if [ ${driveName} = "/dev/sda" ]; then
+        continue
+    fi
+    if [ ${driveName} = "/dev/sdb" ]; then
+        continue
+    fi
+
+    fdisk -l $driveName >fdisk.dat 2>/dev/null
+    # Format the Disk and Create a file system , Mount and create file on it .
+    (echo d;echo;echo w)|fdisk $driveName
+    (echo n;echo p;echo 1;echo;echo;echo w)|fdisk $driveName
+    if [ "$?" = "0" ]; then
+        sleep 5
+
+        # IntegrityCheck $driveName
+        mkfs.ext4 ${driveName}1
+        if [ "$?" = "0" ]; then
+            LogMsg "mkfs.ext4   ${driveName}1 successful..."
+            mount ${driveName}1 /mnt
+            if [ "$?" = "0" ]; then
+                LogMsg "Drive mounted successfully..."
+                mkdir /mnt/Example
+                dd if=/dev/zero of=/mnt/Example/data bs=10M count=50
+                if [ "$?" = "0" ]; then
+                    LogMsg "Successful created directory /mnt/Example"
+                    LogMsg "Listing directory: ls /mnt/Example"
+                    ls /mnt/Example
+                    rm -f /mnt/Example/data
+                    df -h
+                    umount /mnt
+                    if [ "$?" = "0" ]; then
+                        LogMsg "Drive unmounted successfully..."
+                    fi
+                    LogMsg "Disk test's completed for ${driveName}1"
+                else
+                    LogMsg "Error in creating directory /mnt/Example..."
```

Comment:
LogErr for all these failures and below?

Reply:
Updated